lisa/sut_orchestrator/azure/common.py

```diff
@@ -1036,6 +1048,40 @@ def delete_file_share(
     share_service_client.delete_share(file_share_name)
 
 
+def save_console_log(
+    resource_group_name: str,
+    vm_name: str,
+    platform: "AzurePlatform",
+    log: Logger,
+    saved_path: Optional[Path],
+    screenshot_file_name: str = "serial_console",
+) -> bytes:
+    diagnostic_data = _get_diagnostic_data(platform, resource_group_name, vm_name)
+    if saved_path:
+        screenshot_raw_name = saved_path.joinpath(f"{screenshot_file_name}.bmp")
+        screenshot_name = saved_path.joinpath(f"{screenshot_file_name}.png")
+        screenshot_response = requests.get(diagnostic_data.console_screenshot_blob_uri)
+        with open(screenshot_raw_name, mode="wb") as f:
+            f.write(screenshot_response.content)
+        try:
+            with Image.open(screenshot_raw_name) as image:
+                image.save(screenshot_name, "PNG", optimize=True)
+        except UnidentifiedImageError:
+            log.debug(
+                "The screenshot is not generated. "
+                "The reason may be the VM is not started."
+            )
+        unlink(screenshot_raw_name)
```

Comment:
Simple changes to leverage pathlib Paths

```suggestion
    if saved_path:
        screenshot_raw_name = saved_path / f"{screenshot_file_name}.bmp"
        screenshot_response = requests.get(diagnostic_data.console_screenshot_blob_uri)
        screenshot_raw_name.write_bytes(screenshot_response.content)

        try:
            with Image.open(screenshot_raw_name) as image:
                image.save(saved_path / f"{screenshot_file_name}.png", "PNG", optimize=True)
        except UnidentifiedImageError:
            log.debug(
                "The screenshot is not generated. "
                "The reason may be the VM is not started."
            )
        screenshot_raw_name.unlink()
```

Reply:
updated